### PR TITLE
Instant Search: Show unicode in URL path

### LINKED ISSUE #16738: Search: results are displaying non-ASCII characters as encoded in the URL path
Unsure if we need to fix this in the API or in JS.

Screen Shot 2020-08-05 at 15 22 16 (1)

### PR DESCRIPTION
**Pull request by jsnmoon**
Fixes #16738.

#### Changes proposed in this Pull Request:
* Decodes encoded path pieces to show intelligible URLs.

> Before
> Screen Shot 2020-10-20 at 3 08 32 PM

> After
> Screen Shot 2020-10-20 at 3 08 03 PM

#### Does this pull request change what data or activity we track or use?
None.

#### Testing instructions:
1. Apply this patch to your site with Jetpack Search enabled.
2. Set your search result style to "Expanded".
3. Create a page with a non-ASCII path.
4. Search for the page in step 2. Ensure that the URL appears intelligible.

#### Proposed changelog entry for your changes:
* Improved URL formatting for the expanded search layout.

### CODE REVIEW

**Comment by bluefuton** [comment]
Works well for the "expanded" result format:

Screen Shot 2020-10-21 at 16 21 26

Shall we fix it for the "minimal" result format too?

**Comment by jsnmoon** [comment]
> Shall we fix it for the "minimal" result format too?

Done! I went and componentized the path pieces into PathBreadcrumbs. This is ready for another round of team reviews.

I'd like to wait until #17495 lands before a crew review, however. Ideally, the new PathBreadcrumbs component would use Webpack for bundling its styles.

**Comment by jsnmoon** [comment]
Rebased for good measure.

File: modules/search/instant-search/components/path-breadcrumbs.scss
Comment (bluefuton) [inline modules/search/instant-search/components/path-breadcrumbs.scss:1]: Do we need to include this stylesheet somewhere? I'm not seeing these styles applied:

Screen Shot 2020-10-27 at 17 13 49

Comment (jsnmoon) [reply modules/search/instant-search/components/path-breadcrumbs.scss:3]: Whoops, I botched the stylesheet import with a typo. This has been fixed.

**Comment by jsnmoon** [comment]
Rebased for good measure.

**Review by bluefuton — APPROVED** [review]
Works well 🚢 

Screen Shot 2020-10-28 at 13 42 33

**Review by jeherve — APPROVED** [review]
Works well in my tests

screenshot 2020-10-28 at 13 10 31

**Pull request merged**
